### DISCUSSION TITLE
Plugins at runtime or import time?

### DISCUSSION THREAD

**Discussion by Jaapel** [Deltares/hydromt#865] [Feature requests]
Currently plugins are checked when you search for them, e.g.
```python
Plugin().driver_plugins
```
If you make a mistake in your driver definition you only find out during runtime. If we load all plugins during import time, this could help find mistakes early at the cost of import time.
What is desirable?

**Comment by aslowwriter** [comment]
This actually is very hard to do at import time, since if we want to do that we'll have to be much more strict on the import order, since for the plugins to be registered, the imports need to have happened. I agree that it would be nice to have this validation, but it will impose more constrains on how we manage imports.
